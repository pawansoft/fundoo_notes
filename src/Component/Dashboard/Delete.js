import React, { Component } from 'react';
import {
    View,
    ScrollView,
    Text,
} from 'react-native';
import AsyncStorage from '@react-native-async-storage/async-storage';
import FirebaseService from '../../../Services/firebase_services/NoteServices';
import { Card, Modal, Paragraph, Portal, Title, Provider, Menu, Appbar, Button } from 'react-native-paper';
import NotesContainerStyle from '../../Style/NotesContainerStyle';
import ProfileStyle from '../../Style/ProfileStyle';
import Profile from './Profile';
import RBSheet from 'react-native-raw-bottom-sheet';
import dashboardStyle from '../../Style/dashboardStyle';

export default class Delete extends Component {
    constructor(props) {
        super(props)
        this.state = {
            listView: true,
            notes: [],
            showProfileScreen: false
        }
    }

    showProfile = async () => {
        await this.setState({
            showProfileScreen: true
        })
    }

    hideProfile = async () => {
        await this.setState({
            showProfileScreen: false
        })
    }

    async componentDidMount() {
        const userid = await AsyncStorage.getItem('userId');
        await FirebaseService._getNoteService(userid).then(async data => {
            let notes = data ? data : {}

            await this.setState({
                notes: notes
            })
        })
    }

    selectView = async () => {
        await this.setState({
            listView: !this.state.listView
        })
        console.log(this.state.listView);
    }

    showDeleteSnackbar = async() =>{
        if(this.props.route.params != undefined) {
            if(this.props.route.params.isEmpty != undefined) {
            await this.setState({
                showEmptyNoteSnackbar : true
            })
            }
        }
    }

    handleRBSheetOpenButton = async () => {
        this.RBSheet.open();
    }

    handleCancel = () => {
        const {onPress} = this.props
        this.RBSheet.close()
        //onPress();
    }

    deleteNoteActionHandler = (key) => {
        this.props.navigation.navigate('DeleteAction', 
        { key: key, notes: this.state.notes[key], navigation : this.props.navigation })
    }

    render() {
        let noteKey = Object.keys(this.state.notes);
        return (
            <Provider>
            <View style={{ flex: 1 }}>
                <View>
                <Appbar style = {dashboardStyle.headerContainer}>
                    <Appbar.Action
                    icon = 'menu'
                    onPress = {() => this.props.navigation.openDrawer()}/>
                <Text>Deleted Notes</Text>            
                <Appbar.Action
                    style = {{marginRight : 10}}/>
                </Appbar>
                </View>
                <ScrollView>
                    <View style={{ flexDirection: 'row', flexWrap: 'wrap' }}>
                        {noteKey.length > 0 ?
                            noteKey.reverse().map(key => (
                                <React.Fragment key={key}>
                                    {this.state.notes[key].NotesDetail.isDeleted ?
                                        (<Card
                                            onPress = {() => this.deleteNoteActionHandler(key)}
                                            style={(this.state.listView) ? NotesContainerStyle.container_list : NotesContainerStyle.container}>
                                            <Card.Content style = {{backgroundColor: 'white'}}>
                                                <Title style = {{color: 'black'}}>
                                                    {this.state.notes[key].NotesDetail.title}
                                                </Title>
                                                <Paragraph style = {{color: 'black'}}>
                                                    {this.state.notes[key].NotesDetail.note}
                                                </Paragraph>
                                            </Card.Content>
                                        </Card>)
                                        : null}
                                </React.Fragment>
                            ))
                            :
                            null
                        }
                    </View>
                </ScrollView>
                <Portal>
                    <Modal
                        visible={this.state.showProfileScreen}
                        onDismiss={this.hideProfile}
                        contentContainerStyle = {ProfileStyle.container}>
                        <Profile navigation={this.props.navigation} />
                    </Modal>
                </Portal>
            </View>
            
            </Provider>
        )
    }
}