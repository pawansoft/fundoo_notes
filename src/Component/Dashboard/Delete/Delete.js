import React, { Component } from 'react';
import {
    View,
    ScrollView,
    Text,
    Image,
} from 'react-native';
import AsyncStorage from '@react-native-async-storage/async-storage';
import FirebaseService from '../../../../Services/firebase_services/NoteServices';
import { Card, Paragraph, Title, Provider, Appbar, Button } from 'react-native-paper';
import NotesContainerStyle from '../../../Style/NotesContainerStyle';
import dashboardStyle from '../../../Style/dashboardStyle';
import RBSheet from 'react-native-raw-bottom-sheet';
import SQLiteCRUDService from '../../../../Services/SQLite_service/SQLiteCRUDService';
import backgroundImageStyle from '../../../Style/backgroundImageStyle';
import SQLiteLabelServices from '../../../../Services/SQLite_service/SQLiteLabelServices';
import NotesHolderStyle from '../../../Style/NotesHolderStyle';

export default class Delete extends Component {
    constructor(props) {
        super(props)
        this.state = {
            listView: true,
            notesFromSQLite: [],
            showProfileScreen: false,
            isEditable: false,
            labelDetails: []
        }
    }

    showProfile = async () => {
        await this.setState({
            showProfileScreen: true
        })
    }

    hideProfile = async () => {
        await this.setState({
            showProfileScreen: false
        })
    }

    async componentDidMount() {

        await SQLiteCRUDService.getConditionalDetailsFromSQLiteDatabase('false', 'true')
            .then(async (data) => {
                var temp = []
                if (data.rows.length != 0) {
                    for (let i = 0; i < data.rows.length; ++i)
                        temp.push(data.rows.item(i));
                    await this.setState({
                        notesFromSQLite: temp
                    })
                }
            })
        await SQLiteLabelServices.selectLabelFromSQliteStorage()
            .then(async result => {
                var temp = [];
                if (result.rows.length != 0) {
                    for (let i = 0; i < result.rows.length; ++i)
                        temp.push(result.rows.item(i));
                    this.setState({
                        labelDetails: temp
                    })
                }
            })
    }

    selectView = async () => {
        await this.setState({
            listView: !this.state.listView
        })
    }

    showDeleteSnackbar = async () => {
        if (this.props.route.params != undefined) {
            if (this.props.route.params.isEmpty != undefined) {
                await this.setState({
                    showEmptyNoteSnackbar: true
                })
            }
        }
    }

    handleRBSheetOpenButton = async () => {
        this.RBSheet.open();
    }

    handleCancel = () => {
        const { onPress } = this.props
        this.RBSheet.close()
        //onPress();
    }

    deleteNoteActionHandler = (value) => {
        this.props.navigation.navigate('DeleteAction',
            { notes: value })
    }

    render() {
        return (
            <Provider>
                <View style={{ flex: 1 }}>
                    <Image style={backgroundImageStyle.backgroundImage} source={require('../../../assets/background1.jpg')}>
                    </Image>
                    <View>
                        <Appbar style={dashboardStyle.headerContainer}>
                            <Appbar.Action
                                icon='menu'
                                onPress={() => this.props.navigation.openDrawer()} />
                            <Text>Deleted Notes</Text>
                            <Appbar.Action
                                icon='dots-vertical'
                                style={{ marginRight: 10 }}
                                onPress={this.handleRBSheetOpenButton} />
                        </Appbar>
                    </View>

                    <ScrollView>
                        <View style={{ flexDirection: 'row', flexWrap: 'wrap' }}>
                            {this.state.notesFromSQLite.reverse().map(val => (
                                <React.Fragment key={val.NoteKey}>
                                    {val.isDeleted == "true" ?

                                        (<Card
                                            onPress={() => this.deleteNoteActionHandler(val)}
                                            style={NotesContainerStyle.container_list}>

                                            <Card.Content style={{ backgroundColor: 'white' }}>
                                                <Title style={{ color: 'black' }}>
                                                    {val.Title}
                                                </Title>
                                                <Paragraph style={{ color: 'black' }}>
                                                    {val.Notes}
                                                </Paragraph>
                                                <View style = {NotesHolderStyle.label_text_container}>
                                                    {(val.Labels != undefined) ?
                                                        this.state.labelDetails.map(data =>
                                                            val.Labels.includes(data.label_id) ?
                                                                <React.Fragment key={data.label_id}>
                                                                    <Text style={NotesHolderStyle.label_text}>{data.label}</Text>
                                                                </React.Fragment>
                                                                : null
                                                        )
                                                        : null
                                                    }
                                                </View>
                                            </Card.Content>
                                        </Card>)
                                        : null}
                                </React.Fragment>
                            ))
                            }
                        </View>
                    </ScrollView>
                    <RBSheet
                        ref={ref => { this.RBSheet = ref }}
                        height={100}
                        customStyles={{
                            container: {
                                borderTopWidth: 1,
                                borderColor: "#d3d3d3",
                            },
                            wrapper: {
                                backgroundColor: "transparent",
                            },
                        }}>
                        <View style={{ width: '80%', marginLeft: '10%', marginTop: '10%', borderRadius: 10 }}>
                            <Button
                                icon='delete-outline'>
                                <Text style={{ color: 'Red', fontWeight: 'bold', fontSize: 18 }}>
                                    Clean Recycle bin
                            </Text>
                            </Button>
                        </View>
                    </RBSheet>
                </View>
            </Provider>
        )
    }
}