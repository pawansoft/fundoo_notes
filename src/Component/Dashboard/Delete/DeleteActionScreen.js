import React, { Component } from 'react';
import {
    Image,
    ScrollView,
    Text,
    View
} from 'react-native';
import { Appbar, Button, Menu, Snackbar } from 'react-native-paper';
import RBSheet from 'react-native-raw-bottom-sheet';
import AsyncStorage from '@react-native-async-storage/async-storage';
import NotesHolderStyle from '../../../Style/NotesHolderStyle';
import NotesServiceController from '../../../../Services/data_flow_controller/NotesServiceController';
import backgroundImageStyle from '../../../Style/backgroundImageStyle';
import SQLiteLabelServices from '../../../../Services/SQLite_service/SQLiteLabelServices';
import NotesContainerStyle from '../../../Style/NotesContainerStyle';
import Datalayr from '../../../../Services/Datalayer/Datalayr';

export class DeleteActionScreen extends Component {
    constructor(props) {
        super(props)
        this.state = {
            title: '',
            note: '',
            key: '',
            userid: '',
            isDeletable: false,
            selectedLabel: [],
            labelDetails: []
        }
    }

    componentDidMount = async () => {
        const userId = await AsyncStorage.getItem('userId')
        await this.setState({
            userid: userId
        })

        if (this.props.route.params != undefined) {
            console.log(this.state.key);
            await this.setState({
                key: this.props.route.params.notes.NoteKey,
                note: this.props.route.params.notes.Notes,
                title: this.props.route.params.notes.Title,
                selectedLabel: this.props.route.params.notes.Labels
            })
        }

        await SQLiteLabelServices.selectLabelFromSQliteStorage()
            .then(async result => {
                var temp = [];
                if (result.rows.length != 0) {
                    for (let i = 0; i < result.rows.length; ++i)
                        temp.push(result.rows.item(i));
                    this.setState({
                        labelDetails: temp
                    })
                }
            })
    }

    handleRBSheetOpenButton = async () => {
        this.RBSheet.open();
    }

    handleRestoreService = () => {
        Datalayr.handleRestoreNote(this.props.route.params.notes.NoteKey)
        this.props.navigation.push('Home', { screen: 'Delete' })
    }

    handleDeleteNoteForEver = () => {
        this.handleCancel();
        Datalayr.DeleteNoteService(this.state.key)
        this.props.navigation.push('Home', { screen: 'Delete' })

    }

    handleRBSheetOpenButton = async () => {
        this.RBSheet.open();
    }

    onDismissSnakbarHandler = async () => {
        await this.setState({
            isDeletable: false
        })
    }

    handleOpenSnakBar = async () => {
        await this.setState({
            isDeletable: true
        })
    }

    restoreNotesHandler = () => {
        this.handleCancel();
        Datalayr.handleRestoreNote(this.props.route.params.notes.NoteKey)
        this.props.navigation.push('Home', { screen: 'Notes' })
    }
    handleCancel = () => {
        const { onPress } = this.props
        this.RBSheet.close()
    }

    render() {
        return (
            <View style={{ flex: 1 }}>
                <Image style={backgroundImageStyle.backgroundImage} source={require('../../../assets/background1.jpg')}>
                </Image>
                <Appbar style={{ backgroundColor: 'white', width: '20%' }}>
                    <Appbar.Action
                        icon='keyboard-backspace'
                        onPress={() => this.props.navigation.push('Home', { screen: 'Delete' })}
                    />
                </Appbar>
                <ScrollView>
                    <View style={NotesHolderStyle.Title}>
                        <Text style={NotesHolderStyle.TitleFont}
                            onPress={this.handleOpenSnakBar}>
                            {this.state.title}
                        </Text>
                    </View>
                    <View style={NotesHolderStyle.Title}>
                        <Text>
                            {this.state.note}
                        </Text>
                    </View>
                    <View style={NotesHolderStyle.label_text_container}>
                        {(this.state.selectedLabel != undefined) ?
                            this.state.labelDetails.map(data =>
                                this.state.selectedLabel.includes(data.label_id) ?
                                    <React.Fragment key={data.label_id}>
                                        <Text style={NotesHolderStyle.label_text}>{data.label}</Text>
                                    </React.Fragment>
                                    : null

                            ) : null}
                    </View>
                </ScrollView>
                <View>
                    <Appbar style={{ backgroundColor: 'white', justifyContent: 'space-around' }}>
                        <Appbar.Action
                            icon='plus-box-outline' />
                        <Appbar.Action
                            title='Deleted note' />
                        <Appbar.Action
                            icon='dots-vertical'
                            onPress={this.handleRBSheetOpenButton} />
                    </Appbar>
                </View>
                <RBSheet
                    ref={ref => { this.RBSheet = ref }}
                    height={200}
                    customStyles={{
                        container: {
                            borderTopWidth: 1,
                            borderColor: "#d3d3d3",
                            marginBottom: 50
                        },
                        wrapper: {
                            backgroundColor: "transparent",
                        },
                    }}>
                    <View style={{ marginTop: '10%' }}>
                        <Menu.Item
                            icon='autorenew'
                            onPress={this.handleRestoreService}
                            title=' Restore' />
                        <Menu.Item
                            title='Delete forever'
                            icon='delete-outline'
                            onPress={this.handleDeleteNoteForEver} />

                    </View>
                </RBSheet>
                <Snackbar
                    style={{ marginBottom: '30%', flexDirection: 'row', justifyContent: 'space-around', backgroundColor: 'grey' }}

                    visible={this.state.isDeletable}
                    onDismiss={this.onDismissSnakbarHandler}
                    duration={5000}>
                    <View style={{ flexDirection: 'row', justifyContent: 'space-around' }}>
                        <View>
                            <Text style={{ marginTop: 10, color: 'white' }}>
                                Note Deleted Successfully
                            </Text>
                        </View>

                        <View style={{ marginLeft: 50 }}>
                            <Button
                                onPress={this.restoreNotesHandler}>
                                <Text style={{ color: '#cca300' }}>Restore</Text>
                            </Button>
                        </View>
                    </View>
                </Snackbar>
            </View>
        )
    }
}