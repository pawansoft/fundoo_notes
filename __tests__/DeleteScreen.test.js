import React from 'react';
import{configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
import mockAsyncStorage from '@react-native-async-storage/async-storage/jest/async-storage-mock';
import Delete from '../src/Component/Dashboard/Delete';

jest.mock('@react-native-async-storage/async-storage', () => mockAsyncStorage);
configure({adapter: new Adapter()})

jest.mock('react-native-localization', () => class RNLocalization {
    language = 'en-US'
  
    constructor (props) {
      this.props = props
      this.setLanguage(this.language)
    }
  
    setLanguage (interfaceLanguage) {
      this.language = interfaceLanguage
      if (this.props[interfaceLanguage]) {
        var localizedStrings = this.props[this.language]
        for (var key in localizedStrings) {
          if (localizedStrings.hasOwnProperty(key)) this[key] = localizedStrings[key]
        }
      }
    }
})

describe('Delete screen', () => {
    it('ProvideDeleteScreen_WhenSnapshotMatches_ShouldPassTestCase', () =>{
        const component = shallow(<Delete/>)
        expect(component).toMatchSnapshot();
    })
})