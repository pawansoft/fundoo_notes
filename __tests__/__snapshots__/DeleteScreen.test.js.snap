// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delete screen ProvideDeleteScreen_WhenSnapshotMatches_ShouldPassTestCase 1`] = `ShallowWrapper {}`;
